### ISSUE TITLE
BIOS files are not tied to the region setting

### ISSUE THREAD

**Issue by zomgugoff** [MiSTer-devel/MegaCD_MiSTer#99]
TL;DR Can manually loading BIOS files and/or using unique file names that are bound to the region setting be implemented?

The core requiring the BIOS filename to be the same regardless of region is not an ideal way to handle the BIOS. Having the option to load it manually (as it was in older builds) and/or having each region's BIOS be named individually is far easier to manage. In the current arrangement, I have to create region folders or put a BIOS file in each game folder. I don't like having to dig through multiple region folders to find a game. Also, the system region setting in the BIOS still has to be manually changed, so separated BIOS files doesn't really make things any easier.

**Comment by birdybro** [comment]
It could potentially be changed.

The intended region for each MegaCD game is at 0x200 consistently, J, U, or E, in the first Bin. This would have to be a change in Main however, and use a similar method to how the PSX core handles it. Something like this would be added to the support files in Main_MiSTer for the regions:

```cpp
enum region_t
{
	UNKNOWN = 0,
	JP,
	US,
	EU
};

const char* region_string(region_t region)
{
	switch (region)
	{
		case region_t::JP: return "J";
		case region_t::US: return "U";
		case region_t::EU: return "E";
		default: return "Unknown";
	}
}
```

And then add somewhere to have it read 0x200's byte as ascii, and pass that region string through hps_io, add the region strings to the configuration string, make ioctl_index wider, initialize it to 0 to match the default case from the c++ support file... it'd be a bit of work, the specifics are over my head, but if anyone else wants to do it, here's some of the info.

I know @markun implemented the region autodetection for the PSX core, maybe markun would be interested in tackling this as well?

**Comment by valerino** [comment]
just stumbled into this, and i did this for my personal usage, glad if it helps somehow :) https://github.com/valerino/MiSTer_MegaCD_bios_select
